api/__init__.py

```diff
@@ -71,3 +72,7 @@ def build_collection_response(data, page, per_page, total):
 
 def custom_escape(expression):
     return reduce(lambda x, y: x + "\\" + y if y in ESCAPE_CHARS else x + y, expression, "")
+
+
+def json_response(title, detail, status=status.HTTP_400_BAD_REQUEST):
+    return {"error": title, "detail": detail, "status": status}
```

Comment:
We need to return a `flask_json_response` with the status code. Otherwise it will always return `200` in the response status.

```suggestion
    return flask_json_response({"error": title, "detail": detail, "status": status}, status)
```

Take a look at: https://github.com/RedHatInsights/insights-host-inventory/pull/1418/files#diff-d8438cdb658524a2b6e086de042e2cd85ef375b8584204e05e09fd7a0128ce62